Sources/TuistGenerator/Models/InfoPlist.swift

```diff
@@ -2,30 +2,139 @@ import Basic
+            switch (lhs, rhs) {
+            case let (.string(lhsValue), .string(rhsValue)):
+                return lhsValue == rhsValue
+            case let (.integer(lhsValue), .integer(rhsValue)):
+                return lhsValue == rhsValue
+            case let (.boolean(lhsValue), .boolean(rhsValue)):
+                return lhsValue == rhsValue
+            case let (.dictionary(lhsValue), .dictionary(rhsValue)):
+                return lhsValue == rhsValue
+            case let (.array(lhsValue), .array(rhsValue)):
+                return lhsValue == rhsValue
+            default:
+                return false
+            }
+        }
+    }
+
     case file(path: AbsolutePath)
+    case dictionary([String: Value])
+
+    // MARK: - Equatable
 
     public static func == (lhs: InfoPlist, rhs: InfoPlist) -> Bool {
         switch (lhs, rhs) {
         case let (.file(lhsPath), .file(rhsPath)):
             return lhsPath == rhsPath
+        case let (.dictionary(lhsDictionary), .dictionary(rhsDictionary)):
+            return lhsDictionary == rhsDictionary
+        default:
+            return false
         }
     }
 
-    public var path: AbsolutePath {
+    // MARK: - Public
+
+    public var path: AbsolutePath? {
         switch self {
         case let .file(path):
             return path
+        default:
+            return nil
         }
     }
+}
 
-    // MARK: - ExpressibleByStringLiteral
-
-    public typealias UnicodeScalarLiteralType = String
-    public typealias ExtendedGraphemeClusterLiteralType = String
-    public typealias StringLiteralType = String
+// MARK: - InfoPlist - ExpressibleByStringLiteral
 
+extension InfoPlist: ExpressibleByStringLiteral {
     public init(stringLiteral value: String) {
         self = .file(path: AbsolutePath(value))
     }
 }
+
+// MARK: - InfoPlist.Value - ExpressibleByStringLiteral
+
+extension InfoPlist.Value: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self = .string(value)
+    }
+}
+
+// MARK: - InfoPlist.Value - ExpressibleByIntegerLiteral
+
+extension InfoPlist.Value: ExpressibleByIntegerLiteral {
+    public init(integerLiteral value: Int) {
+        self = .integer(value)
+    }
+}
+
+// MARK: - InfoPlist.Value - ExpressibleByBooleanLiteral
+
+extension InfoPlist.Value: ExpressibleByBooleanLiteral {
+    public init(booleanLiteral value: Bool) {
+        self = .boolean(value)
+    }
+}
+
+// MARK: - InfoPlist.Value - ExpressibleByDictionaryLiteral
+
+extension InfoPlist.Value: ExpressibleByDictionaryLiteral {
+    public init(dictionaryLiteral elements: (String, InfoPlist.Value)...) {
+        self = .dictionary(Dictionary(uniqueKeysWithValues: elements))
+    }
+}
+
+// MARK: - InfoPlist.Value - ExpressibleByArrayLiteral
+
+extension InfoPlist.Value: ExpressibleByArrayLiteral {
+    public init(arrayLiteral elements: InfoPlist.Value...) {
+        self = .array(elements)
+    }
+}
```

Comment:
You don't need to duplicate all of the protocols or Codable conformance inside TuistGenerator, I'll leave it up to you. Removing these causes no problems to the feature working.